src/internal-model.js

```diff
@@ -90,15 +94,17 @@ export default Model.extend({
   },
 
   hasRelated (relationName) {
-    const relationship = this.getRelationship(relationName, false)
+    const related = this.getRelationshipData(relationName, false)
 
-    return !!_.result(relationship, ['data', 'id'])
+    return !!(_.isArray(related)
+      ? related.length
+      : _.result(related, 'id'))
```

Comment:
```js
const prop = _.isArray(related)
  ? 'length'
  : 'id';

return Boolean(_.result(related, prop);
```

Or if `data` is either an empty array or empty object, i.e. not just `id` is missing, then `_.isEmpty` might suffice?

```js
return _.isEmpty(this.getRelationshipData(relationName, false));
```

Reply:
```javascript
hasRelated (relationName) {
  return !_.isEmpty(this.getRelationshipData(relationName, false))
},
```
will work beautifully! `hasRelated` already doesn't check for `type` or that `id` exists in `has-many` Just checking that the relationship data isn't empty should suffice for the requirements of this method.